src/reynir/verbframe.py

```diff
@@ -248,86 +255,162 @@ def matches_pcl(self, particle: str) -> bool:
+            return s, error            
+
+        def get_particle(s: str) -> Tuple[str, Optional[str]]:
+            # Process particles, should only be one in each line
+            particle = None
+            ix = s.rfind("*")
+            if ix >= 0:
+                particle = s[ix:].strip()
+                s = s[0:ix].strip()
+                if " " in particle:
+                    raise ConfigError("Particle should only be one word")
+                elif len(particle) < 2:
+                    raise ConfigError("Particle should be at least one letter")
+            return s, particle
+
+        def get_preposition(s: str) -> Tuple[str, List[Tuple[str, str]]]:
+            # Process preposition arguments, if any
+            prepositions: List[Tuple[str, str]] = []
+            ap = s.split("/")
+            s = ap[0]
+            ix = 1
+            while ix < len(ap):
+                # We expect something like 'af þgf', or possibly
+                # 'fyrir_hönd þf' (where the underscore needs to be replaced by a space)
+                # or 'milli ef_ft' (detailing the plural)
+                p = ap[ix].strip()
+                parg = p.split()
+                #if len(parg) != 2:
+                #    raise ConfigError("Preposition should have exactly one argument")
+                if parg[1] not in ALL_CASES and parg[1] not in SUBCLAUSES:
+                    parg[1] = REFLPRN.get(parg[1], parg[1])
+                    assert parg[1] is not None
+                    spl = parg[1].split("_")
+                    skip = ["gr", "ft", "est", "mst", "et", "kk", "kvk", "hk"]   # TODO Handle these variants
+                    while True:
+                        if spl[-1].strip() in skip:
+                            spl = spl[:-1]
+                        else:
+                            break
+                    if spl[-1] not in ALL_CASES:
+                        raise ConfigError(
+                            "Preposition argument must have a case as its last variant"
+                        )
+                    parg[1] = spl[-1]
+                prepositions.append((parg[0].replace("_", " "), parg[1]))
+                ix += 1
+            return s, prepositions
+
+        def get_direct_object(s: str) -> Tuple[str, str]:
+            # Process direct object argument
+            op = s.split("|")
+            s = op[0]
+            obj: str = ""
+            if len(op) < 1 or len(op) > 2:
+                raise ConfigError("Verb should have zero or one direct object")
+            elif len(op) == 2:
+                pobj = op[1].strip()
+                if pobj not in ALL_CASES and pobj not in SUBCLAUSES:
+                    pobj = pobj.split(" ")[0].strip()
+                    pobj = REFLPRN.get(pobj, pobj)
+                    assert pobj is not None
+                    spl = pobj.split("_")
+                    skip = ["gr", "ft", "est", "mst", "et", "kk", "kvk", "hk"]
```

Comment:
Maybe this variant handling logic could be factored out into a separate function

Reply:
Done, much clearer!